compiler/front_end/BUILD

```diff
@@ -99,22 +99,57 @@ py_library(
         "error_examples",
     ],
     deps = [
+        ":cached_parser",
         ":lr1",
+        ":make_parser",
         ":module_ir",
-        ":tokenizer",
-        "//compiler/util:resources",
+        "//compiler/util:parser_types",
         "//compiler/util:simple_memoizer",
     ],
 )
 
+py_library(
+    name = "cached_parser",
+    srcs = ["cached_parser.py"],
```

Comment:
Not sure how annoying this would be, but we could maybe have this live under a `generated/` sub-dir to just make it clear what it's provenance is.